extensions/blocks/premium-content/_inc/access-check.php

```diff
@@ -26,6 +26,54 @@ function pre_render_checks() {
 	return true;
 }
 
+/**
+ * Determines if the site has a plan that supports the
+ * Premium Content block. If false, the site requires a
+ * plan upgrade.
+ *
+ * @return bool
+ */
+function required_plan_checks() {
+	// For WPCOM sites.
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM && function_exists( 'has_any_blog_stickers' ) ) {
+		$site_id = get_current_blog_id();
+		return has_any_blog_stickers( array( 'personal-plan', 'premium-plan', 'business-plan', 'ecommerce-plan' ), $site_id );
+	}
+
+	// For Jetpack sites, only Atomic sites (with a business plan
+	// or above) have the block, so no upgrade is required.
+	return true;
```

Comment:
This is no longer true, now that the block is moved out of ETK. I think we need something like:

```
return Jetpack::is_active() && Jetpack_Plan::supports( 'premium-content' )
```

This is now extremely similar to the code for [is_enabled_jetpack_recurring_payments](https://github.com/Automattic/jetpack/blob/9e1594c234a14d8b42d6f765b4770f42df3b2c56/modules/memberships/class-jetpack-memberships.php#L446), which is itself similar to [set_availability_for_plan](https://github.com/Automattic/jetpack/blob/9e1594c234a14d8b42d6f765b4770f42df3b2c56/class.jetpack-gutenberg.php#L1011) in the `Jetpack_Gutenberg` class. I think it's a good opportunity for a refactor, although not necessary in this PR.

Reply:
This results in a bug where the Stripe nudge is displayed instead of the Upgrade nudge on free Jetpack sites. Since this won't even be reproducible until we get the block rendering on Jetpack free plans, I will make a follow up ticket for this.